MovieAppPetProject/MovieAppPetProject/GetMovieDetailsByID/GetMovieDetails-ViewModel.swift

```diff
@@ -10,54 +10,48 @@ import Foundation
 class MovieDetailsViewModel {
 
     // MARK: - Variables
-    var movieDetails: MovieDetails?
-    typealias MovieDetailsCompletion = (Result<MovieDetails, CustomError>) -> Void
-    private var selectedSearchMovie: SearchMoviesResults?
-    private var selectedHomeScreenMovies: NowPlayingResults?
-    private let movieDetailsRepository: MovieDetailsRepositoryType?
+    private let movieDetailsRepository: MovieDetailsRepositoryType
+    private let movieID: Int
+    var movieDetails: MovieDetails? {
+        didSet {
+            didUpdateDetails?()
+        }
+    }
 
-    // MARK: - Initializer
-    init(movieDetails: MovieDetails?, selectedSearchMovie: SearchMoviesResults?, selectedHomeScreenMovies: NowPlayingResults?, movieDetailsRepository: MovieDetailsRepositoryType) {
-        self.movieDetails = movieDetails
-        self.selectedSearchMovie = selectedSearchMovie
-        self.selectedHomeScreenMovies = selectedHomeScreenMovies
-        self.movieDetailsRepository = movieDetailsRepository
+    // MARK: - Computed properties
```

Comment:
You've declared functions but marked them as computed variables?

Reply:
will change them 💪